tiledb/sm/misc/comparators.h

```diff
@@ -202,6 +202,57 @@ class GlobalCmp {
   unsigned dim_num_;
 };
 
+/**
+ * Wrapper of comparison function for sorting coords on the global order
+ * of some domain.
+ */
+class GlobalCmp2 {
+ public:
+  /**
+   * Constructor.
+   *
+   * @param domain The array domain.
+   * @param buff The coordinate buffers, one per dimension, containing the
+   *     actual values, used in positional comparisons.
+   */
+  GlobalCmp2(const Domain* domain, const std::vector<const void*>& coord_buffs)
+      : domain_(domain)
+      , coord_buffs_(coord_buffs) {
```

Comment:
This is unsafe because `coord_buffs` is a temporary bound to a class member reference (`coord_buffs_`). The temporary `coord_buffs` is only guaranteed to persist for the lifetime of this constructor. In other words: `coord_buffs_` will NOT extend the lifetime of `coord_buffs`. When `coord_buffs_` is dereferenced outside of the constructor, for example: within `GlobalCmp2::operator()`, it may be invalid.

From the C++03 standard:
```
The second context is when a reference is bound to a temporary. The temporary to which the reference is bound or the temporary that is the complete object to a subobject of which the temporary is bound persists for the lifetime of the reference except as specified below. A temporary bound to a reference member in a constructor’s ctor-initializer (§12.6.2 [class.base.init]) persists until the constructor exits.
```

So we must ensure that the original `coord_buffs` outlives this `GlobalCmp2`.  
I think the right thing to do here is to change `coord_buffs_` from a const-ref to a const raw ptr. The contract is the same, but if the object is freed before we access it, we'll get a segfault instead of a garbage object.

Or, we could copy/move it, but I don't recommend that.